core/src/main/java/org/axonframework/config/EventHandlingConfiguration.java

```diff
@@ -234,13 +252,25 @@ public EventHandlingConfiguration registerTrackingProcessor(String name,
      * @param processorConfiguration The configuration for the processor
      * @param sequencingPolicy       The sequencing policy to apply when processing events in parallel
      * @return this EventHandlingConfiguration instance for further configuration
+     * @deprecated use {@link EventProcessorRegistry#registerTrackingEventProcessor(String, Function, Function)} instead
      */
+    @Deprecated
     public EventHandlingConfiguration registerTrackingProcessor(String name,
                                                                 Function<Configuration, StreamableMessageSource<TrackedEventMessage<?>>> source,
                                                                 Function<Configuration, TrackingEventProcessorConfiguration> processorConfiguration,
                                                                 Function<Configuration, SequencingPolicy<? super EventMessage<?>>> sequencingPolicy) {
-        return registerEventProcessor(name, (conf, n, handlers) ->
-                buildTrackingEventProcessor(conf, name, handlers, processorConfiguration, source, sequencingPolicy));
+        eventProcessorBuilders.put(name,
+                                   (conf, handlers) ->
+                                           conf.eventProcessorRegistry()
+                                               .registerEventProcessor(name,
+                                                                       (n, c, ehi) ->
+                                                                               buildTrackingEventProcessor(c,
```

Comment:
I had assumed all public calls from this class to delegate to the `EventProcessorRegistry`, but you're still using the private `buildTrackingEventProcessor()` here. 
How come?

Reply:
This one is because of backward compatibility - since (as I mentioned earlier) `EventProcessorRegistry` is not responsible for creation of `EventHandlerInvoker`s, hence if someone used `sequencingPolicy` to configure `eventHandlingInformation`, we would lose that information it cause unexpected behavior. Hence a bit of old logic is left in here.

Reply:
A ha, that makes sense.